src/server.rs

```diff
@@ -0,0 +1,39 @@
+use async_graphql::http::{playground_source, GraphQLPlaygroundConfig};
+use async_graphql_axum::GraphQL;
+use axum::{
+    response::{Html, IntoResponse},
+    routing::get,
+    Router,
+};
+use log::info;
+use sea_orm::DatabaseConnection;
+
+use crate::{configuration::Api, schema::build_schema};
+
+pub(crate) async fn graphql_playground() -> impl IntoResponse {
+    Html(playground_source(
+        GraphQLPlaygroundConfig::new("/").subscription_endpoint("/ws"),
+    ))
+}
+
+pub async fn serve(api: &Api, database: DatabaseConnection) {
+    let schema = build_schema(api.depth_limit, api.complexity_limit, database);
+
+    let app = Router::new().route(
+        "/",
+        get(graphql_playground).post_service(GraphQL::new(schema)),
+    );
+
+    let socket_addr = format!("{}:{}", api.host, api.port)
+        .parse()
+        .expect("Failed to parse socket address");
+
+    tokio::spawn(async move {
+        axum::Server::bind(&socket_addr)
```

Comment:
I think you should be able to pass in a tuple
```suggestion
        axum::Server::bind(&(api.host, api.port))
```

Reply:
I don't think `bind` accepts those arguments. Did you happen to mistaken `axum` for `actix-web`?

Reply:
I think I thought they had the `Into<SocketAddr>` but seems we need to add an into.
```diff
diff --git a/src/configuration.rs b/src/configuration.rs
index fb9a589..e86219f 100644
--- a/src/configuration.rs
+++ b/src/configuration.rs
@@ -1,3 +1,5 @@
+use std::net::Ipv4Addr;
+
 use config::{Config, Environment, File};
 use serde::Deserialize;
 
@@ -8,16 +10,16 @@ pub struct Ingestion {
     pub polling_interval: u64,
 }
 
-#[derive(Debug, Default, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Deserialize, PartialEq, Eq)]
 pub struct Api {
-    pub host: String,
+    pub host: Ipv4Addr,
     pub port: u16,
 
     pub depth_limit: usize,
     pub complexity_limit: usize,
 }
 
-#[derive(Debug, Default, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Deserialize, PartialEq, Eq)]
 pub struct Configuration {
     pub quasar_database_url: Option<String>,
     pub stellar_node_database_url: Option<String>,
diff --git a/src/server.rs b/src/server.rs
index 31d6c91..758195b 100644
--- a/src/server.rs
+++ b/src/server.rs
@@ -24,9 +24,7 @@ pub async fn serve(api: &Api, database: DatabaseConnection) {
         get(graphql_playground).post_service(GraphQL::new(schema)),
     );
 
-    let socket_addr = format!("{}:{}", api.host, api.port)
-        .parse()
-        .expect("Failed to parse socket address");
+    let socket_addr = (api.host, api.port).into();
 
     tokio::spawn(async move {
         axum::Server::bind(&socket_addr)
```
This works. WDYT?

Reply:
I've applied your suggestion